Add myStrom LED Strip support extending rgblamp bulb_type

asansano opened the pull request:
<!--
  You are amazing! Thanks for contributing to our project!
  Please, DO NOT DELETE ANY TEXT from this template! (unless instructed).
-->


## Proposed change
<!--
  Describe the big picture of your changes here to communicate to the
  maintainers why we should accept this pull request. If it fixes a bug
  or resolves a feature request, be sure to link to that issue in the
  additional information section.
-->
Add myStrom support for LED Strip. Strip has almost the same API call than the existent 'rgblamp' bulb.
Except the HSV Color detection, this is excluded from the bulb_type 'Strip'.


## Type of change
<!--
  What type of change does your PR introduce to Home Assistant?
  NOTE: Please, check only 1! box!
  If your PR requires multiple boxes to be checked, you'll most likely need to
  split it into multiple PRs. This makes things easier and faster to code review.
-->

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New integration (thank you!)
- [ x] New feature (which adds functionality to an existing integration)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Additional information
<!--
  Details are important, and help maintainers processing your PR.
  Please be sure to fill out additional details, if applicable.
-->

- This PR fixes or closes issue: fixes #
- This PR is related to issue: 
- Link to documentation pull request: 

## Checklist
<!--
  Put an `x` in the boxes that apply. You can also fill these out after
  creating the PR. If you're unsure about any of them, don't hesitate to ask.
  We're here to help! This is simply a reminder of what we are going to look
  for before merging your code.
-->

- [x ] The code change is tested and works locally.
- [x ] Local tests pass. **Your PR cannot be merged unless tests pass**
- [x ] There is no commented out code in this PR.
- [x ] I have followed the [development checklist][dev-checklist]
- [x] The code has been formatted using Black (`black --fast homeassistant tests`)
- [x ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

If the code communicates with devices, web services, or third-party tools:

- [ ] The [manifest file][manifest-docs] has all fields filled out correctly.  
      Updated and included derived files by running: `python3 -m script.hassfest`.
- [ ] New or updated dependencies have been added to `requirements_all.txt`.  
      Updated by running `python3 -m script.gen_requirements_all`.
- [ ] For the updated dependencies - a link to the changelog, or at minimum a diff between library versions is added to the PR description.
- [ ] Untested files have been added to `.coveragerc`.

The integration reached or maintains the following [Integration Quality Scale][quality-scale]:
<!--
  The Integration Quality Scale scores an integration on the code quality
  and user experience. Each level of the quality scale consists of a list
  of requirements. We highly recommend getting your integration scored!
-->

- [x ] No score or internal
- [ ] 🥈 Silver
- [ ] 🥇 Gold
- [ ] 🏆 Platinum

<!--
  This project is very active and we have a high turnover of pull requests.

  Unfortunately, the number of incoming pull requests is higher than what our
  reviewers can review and merge so there is a long backlog of pull requests
  waiting for review. You can help here!
  
  By reviewing another pull request, you will help raise the code quality of
  that pull request and the final review will be faster. This way the general
  pace of pull request reviews will go up and your wait time will go down.
  
  When picking a pull request to review, try to choose one that hasn't yet
  been reviewed.

  Thanks for helping out!
-->

To help with the load of incoming pull requests:

- [x] I have reviewed two other [open pull requests][prs] in this repository.

[prs]: https://github.com/home-assistant/core/pulls?q=is%3Aopen+is%3Apr+-author%3A%40me+-draft%3Atrue+-label%3Awaiting-for-upstream+sort%3Acreated-desc+review%3Anone

<!--
  Thank you for contributing <3

  Below, some useful links you could explore:
-->
[dev-checklist]: https://developers.home-assistant.io/docs/en/development_checklist.html
[manifest-docs]: https://developers.home-assistant.io/docs/en/creating_integration_manifest.html
[quality-scale]: https://developers.home-assistant.io/docs/en/next/integration_quality_scale_index.html
[docs-repository]: https://github.com/home-assistant/home-assistant.io

asansano committed:
feat: add support for mystrom led strip type 

the mystrom led strip type is almost equal to the bulb api except the status on color HSV. in this version we extent the bulb type rgblamp with strip and exclude the color detection for this new type strip.

asansano committed:
Merge pull request #1 from Innobyte-GmbH/add-support-for-mystrom-led-strip

feat: add support for mystrom led strip type

asansano committed:
chore: enhanced error logging for myStrom Strip

Enhanced the error logging in case the bulb_type does not match rgblamp nor strip

homeassistant commented:
Hi @asansano, 

It seems you haven't yet signed a CLA. Please do so [here](https://home-assistant.io/developers/cla_sign_start/?pr=home-assistant&#x2F;core#57322).

Once you do that we will be able to review and accept this pull request.

Thanks!

Schnabulation reviewed (approved):
As I own a MyStrom LED strip I have tested this change and can confirm it to be working. However it seems that when the LED strip is manipulated from another application (like the stock MyStrom app) the status in Home Assistant only changes between "on" and "off". But it does not read the color or brightness information.

asansano committed:
Update included back brightness and color

Tested MyStrom Light Strip back again on a HA core-2021.10.5 and it looks that colors and brightness seems to work now as well for LED strip without error.

asansano committed:
Merge branch 'home-assistant:dev' into dev

asansano committed:
Update light.py

asansano committed:
Update light.py

frenck reviewed (changes requested):
@asansano Please run `black` to fix the formatting of the code, to make the build pass. I'm happy to help, but you have disabled access for maintainers to your PR.

asansano committed:
run black on changes

asansano committed:
Merge branch 'home-assistant:dev' into dev

asansano committed:
Merge branch 'home-assistant:dev' into dev

bdraco reviewed (dismissed):
Please address the pylint checks

asansano committed:
refactor: changing of IF statement into IN list statement

refactoring of IF statement into IN list statement based on pylint checks

asansano committed:
Merge branch 'home-assistant:dev' into dev

asansano committed:
refactor: if statement & logg msg

run black on it

frenck reviewed (approved):
Thanks, @asansano 👍
